csrc/disjoint_set.h

```diff
@@ -226,17 +338,23 @@ class DisjointSets {
   }
 
   // Initializes a new set for provided entry
-  //
-  // TODO: Return iterator
-  void initializeSet(T entry) {
-    if (disjoint_set_maps_.find(entry) != disjoint_set_maps_.end()) {
-      return;
+  std::pair<
+      typename std::unordered_map<
+          T,
+          std::shared_ptr<VectorOfUniqueEntries<T, Hash>>,
+          Hash>::iterator,
```

Comment:
nit: Should we do a
```C++
using ItemToSetMap = std::unordered_map<T, .....>
```
somewhere in the class?

Reply:
Added `using DisjointSetMap = ...`